src/components/AppLayout/Header/components/ProviderDetails/PairingDetails.tsx

```diff
@@ -28,10 +28,14 @@ const qrRefresh: CSSProperties = {
 }
 
 const PairingDetails = ({ classes }: { classes: Record<string, string> }): ReactElement => {
+  const onboardUri = getOnboardInstance().getState().wallet.provider?.wc?.uri
```

Comment:
I'd move this to the `getPairingUri` function.

Reply:
You could even go one step further and make a hook for it as well.

Reply:
done

Reply:
Good idea with the hook!